one_fm/api/v1/leave_application.py

```diff
@@ -329,14 +329,14 @@ def fetch_leave_approver(employee: str) -> str:
     Returns:
         str: user id of leave approver
     """
-
+    reports_to = frappe.get_value("Employee", employee, ["reports_to"])     
     employee_shift = frappe.get_list("Shift Assignment",fields=["*"],filters={"employee":employee}, order_by='creation desc',limit_page_length=1)
-    if employee_shift[0].shift:
+    if reports_to:
+        approver = frappe.get_value("Employee", reports_to, ["user_id"])
+    else if employee_shift[0].shift:
         approver, Role = get_action_user(employee,employee_shift[0].shift)
     else:
-        reports_to = frappe.get_value("Employee", employee, ["reports_to"])
-        approver = frappe.get_value("Employee", reports_to, ["user_id"])
-
+        frappe.throw(_("Please assign an approver!!"))
```

Comment:
If we throw from here then the approver will not be return to the api to create leave application.
If the approver returned to https://github.com/ONE-F-M/One-FM/blob/9fd9d05bc0c3698caa6dcc86e1d471ff675da276/one_fm/api/v1/leave_application.py#L265 here then only this line will execute.

Reply:
Okay.